finality-provider/store/storedfp_test.go

```diff
@@ -0,0 +1,124 @@
+package store_test
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/babylonlabs-io/finality-provider/finality-provider/proto"
+	"github.com/babylonlabs-io/finality-provider/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestShouldStart(t *testing.T) {
+	tcs := []struct {
+		name           string
+		currFpStatus   proto.FinalityProviderStatus
+		expShouldStart bool
+	}{
+		{
+			"Created: Should NOT start",
+			proto.FinalityProviderStatus_CREATED,
+			false,
+		},
+		{
+			"Slashed: Should NOT start",
+			proto.FinalityProviderStatus_SLASHED,
+			false,
+		},
+		{
+			"Inactive: Should start",
+			proto.FinalityProviderStatus_INACTIVE,
+			true,
+		},
+		{
+			"Registered: Should start",
+			proto.FinalityProviderStatus_REGISTERED,
+			true,
+		},
+		{
+			"Active: Should start",
+			proto.FinalityProviderStatus_ACTIVE,
+			true,
+		},
+	}
+
+	r := rand.New(rand.NewSource(10))
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			fp := testutil.GenRandomFinalityProvider(r, t)
+			fp.Status = tc.currFpStatus
+
+			shouldStart := fp.ShouldStart()
+			require.Equal(t, tc.expShouldStart, shouldStart)
+		})
+	}
+}
+
+func TestShouldSyncStatusFromVotingPower(t *testing.T) {
```

Comment:
I was thinking maybe we don't need `ShouldSyncStatusFromVotingPower` at all as the cases are covered by `UpdateFpStatusFromVotingPower`. Keeping it is bug-prune as we need to carefully list all the cases where we should skip status sync and we could easily miss some cases.

Reply:
I kept the func `ShouldSyncStatusFromVotingPower` as the `UpdateFpStatusFromVotingPower` modifies the store and I also added a test case if it tries to update the status from slashed, it errors out 
 
 While the `ShouldSyncStatusFromVotingPower` only returns if it should update or not, let me know if you agree to keep both

Reply:
I understand that having `ShouldSyncStatusFromVotingPower` can reduce unnecessary db write but the write cost is negligible. However, having this check increases code complexity and is bug-prune -- we need to do comprehensive tests and still cannot be sure we have listed all the cases -- for example, if we add `jailed` status to fp, we have to consider this case in both `ShouldSyncStatusFromVotingPower` and `UpdateFpStatusFromVotingPower`

Reply:
removed at https://github.com/babylonlabs-io/finality-provider/pull/58/commits/9e8196dc673663eb0e3cb333349067b64e2eb93e, 
Made the log just if the status is changed, also added special case for slashed status at `UpdateFpStatusFromVotingPower`